experimental/oak_baremetal_kernel/src/memory.rs

```diff
@@ -26,7 +26,9 @@ static ALLOCATOR: LockedHeap = LockedHeap::empty();
 #[cfg(test)]
 static ALLOCATOR: LockedHeap = LockedHeap::empty();
 
-pub fn init_allocator<E: E820Entry>(e820_table: &[E]) -> Result<(), &str> {
+pub fn init_allocator<E: boot::E820Entry, B: boot::BootInfo<E>>(
```

Comment:
Could you add a note about how / why you ended up with this version? I think it will be a good learning opportunity for future readers / maintainers.

Reply:
Good point, done.